### ISSUE TITLE
SyntaxError: Unexpected token when overriding version

### ISSUE THREAD

**Issue by eduardomhg** [Microsoft/vsts-extension-build-release-tasks#30]
When setting up a very simple Extension Build, when I set the override version to 0.0.2 (for example) I get the following:

`Error occurred while updating tasks version: Error determining tasks manifest paths: SyntaxError: Unexpected token`

All other fields have the default values, and it works if no override version is set.

**Comment by jessehouwing** [comment]
Any chance I could get a copy of your repo to test? It looks like a parse
issue on one of your task.json files and I agree the error message could be
clearer. Having access to the repo, attaching a zip of the repo to this
issue or a copy in my email (jhouwing at xpirit.com) would make it a lot
easier to debug the issue and get you back in business.

**Comment by eduardomhg** [comment]
The code at https://github.com/eduardomhg/VSTeamServicesExtensions should cause the issue, please let me know if it doesn't.

**Comment by eduardomhg** [comment]
Following your comments, I have unchecked the "override task version" option, and now it works, so the issue is related to that.

**Comment by jessehouwing** [comment]
Submitted a pull request to your project. There are a few strange unicode whitespace characters in your vss-extension.json as well as a unicode Byte-order-mark. After removing those the task works as it should it seems.

**Comment by jessehouwing** [comment]
I'd love it if you'd be able test these changes and if they work, update the review on the marketplace,

**Comment by eduardomhg** [comment]
I don't have time to do it now, but I'm sure it works. I'll confirm it ASAP. Sorry for the confusion, I don't know where those characters came from.

I just updated my review in the market place to 5 stars, thanks for the help!

**Comment by eduardomhg** [comment]
Now that I think of it, the fact that the tfx program itself is able to parse task.json correctly may indicate that your code should be able to parse it as well, I believe.

**Comment by eduardomhg** [comment]
I just investigated the issue and indeed it works with your changes. However, I have simplified the issue to the following: 

- If vss-extension.json is encoded in UTF-8 (no BOM), it works. 
- If it is encoded in UTF-8 with BOM, it fails.

I strongly believe your parser should support the [BOM](https://en.wikipedia.org/wiki/Byte_order_mark).

I have used Notepad++ to easily switch from one encoding to the other.
